src/excmds.ts

```diff
@@ -4023,15 +4030,41 @@ export function autocontain(...args: string[]) {
     if (urlMode || saneMode) {
         args.splice(0, 1)
     }
-    if (args.length !== 2) throw new Error("syntax: autocontain [-{u,s}] pattern container")
+    if (args.length < 2) throw new Error("syntax: autocontain [-{u,s}] pattern container proxy1 proxy2")
 
-    let [pattern, container] = args
+    let [pattern, container, ...proxies] = args
 
     if (!urlMode) {
         pattern = saneMode ? `^https?://([^/]*\\.|)${pattern}/` : `^https?://[^/]*${pattern}/`
     }
 
-    return config.set("autocontain", pattern, container)
+    return config.set("autocontain", pattern, proxies.length ? [container, proxies.join(",")].join("+") : container)
+}
+
+/** Add a proxy.
+    Examples:
+    - `proxyadd socksName socks://hostname:port`
+    - `proxyadd socks4 socks4://hostname:port`
+    - `proxyadd https https://username:password@hostname:port`
```

Comment:
```suggestion
    - `proxyadd https https://username:password@hostname:port`
    - `proxyadd work https://admin:hunter2@bigcorp.example:1337`
```

I think this would make it clearer at a glance that the name can be human-friendly and doesn't have to be the name of the protocol :)